resetcloser.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"io"
+)
+
+type ResetCloser struct {
+	c io.Closer
+}
+
+func NewResetCloser(c io.Closer) *ResetCloser {
+	return &ResetCloser{c: c}
+}
+
+func (rc *ResetCloser) Reset() {
+	rc.c = nil
+}
+
+func (rc *ResetCloser) Close() error {
+	if rc.c == nil {
+		return nil
+	}
+	err := rc.c.Close()
+	if rc.c != nil {
```

Comment:
Not sure how I missed that, thanks for catching it. I'm going to change this slightly so that it nils out the owned `io.Closer` unconditionally. The reason is that this type is designed to allow multiple calls to `Close()`, and if the first one results in an error, I don't want it to get accidentally called again. Feel free to unresolve this conversation, or start a new one, if you think this is the wrong thing to do.